Detectors/GLOQC/src/MatchITSTPCQC.cxx

```diff
@@ -373,24 +394,42 @@ void MatchITSTPCQC::initDataRequest()
 
   // initialize data request, if it was not already done
 
+  if (mDoK0QC) {
+    mSrc = GID::getSourcesMask("ITS,TPC,ITS-TPC,ITS-TPC-TOF,TPC-TOF,TPC-TRD,ITS-TPC-TRD,TPC-TRD-TOF,ITS-TPC-TOF,ITS-TPC-TRD-TOF");
+  }
   mSrc &= mAllowedSources;
 
   if (mSrc[GID::Source::ITSTPC] == 0 || mSrc[GID::Source::TPC] == 0 || mSrc[GID::Source::ITS] == 0) {
     LOG(fatal) << "We cannot do ITSTPC QC, some sources are missing, check sources in " << mSrc;
   }
 
   mDataRequest = std::make_shared<o2::globaltracking::DataRequest>();
+  LOG(info) << "Requesting tracks...";
   mDataRequest->requestTracks(mSrc, mUseMC);
+  LOG(info) << "... done requesting tracks";
+  if (mDoK0QC) {
+    mDataRequest->requestPrimaryVertices(mUseMC);
+    mDataRequest->requestSecondaryVertices(mUseMC);
+  }
 }
 
 //__________________________________________________________
 
 void MatchITSTPCQC::run(o2::framework::ProcessingContext& ctx)
 {
 
+  LOG(info) << "Starting";
+
   // Getting the B field
   mBz = o2::base::Propagator::Instance()->getNominalBz();
 
+  // Getting the SVertexer config params
+  if (mTimestamp == -1 && mDoK0QC) {
+    // we have not yet initialized the SVertexer params; let's do it
+    ctx.inputs().get<o2::vertexing::SVertexerParams*>("SVParam");
```

Comment:
This is QC, the inputs are defined in the json.

Reply:
The part of the finaliseCCDB is not necessary for ConfigurableParams since they do not require any assignment etc. But can be called to e.g. acknowledge or print the object.